pydoctor/model.py

```diff
@@ -693,6 +694,8 @@ def __init__(self, options: Optional['Options'] = None):
             self.extensions = list(extensions.get_extensions())
         assert isinstance(self.extensions, list)
         assert isinstance(self.custom_extensions, list)
+        # pydoctor.astbuilder includes some required extensions, so always add it.
```

Comment:
This design allows us to separate the logic for new features that only depends on the current ast builder flow to be separated in different classes.